opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -232,61 +289,278 @@ message MetricDescriptor {
     //   12. The 1 second collection cycle ends. A metric is exported for the
     //      number of requests received over the interval of time t_1 to
     //      t_0+1 with a value of 1.
-    CUMULATIVE = 3;
+    //
+    // Note that the first value in a sequence of CUMULATIVE metrics
+    // after a reset is equivalent in value to a DELTA metric for the
+    // period since the reset.  Although a CUMULATIVE metric could
+    // technically be reset after every collection event and still be
+    // described as CUMULATIVE, exporters should use DELTA if there is
+    // no intention of repeating the StartTimeUnixNano timestamp.
+    CUMULATIVE = 0x4;
+
+    // The two Structure kinds are exclusive of one another.  One
+    // of the Structure kind elements MUST be set.
+
+    // ADDING structure means the measurement determines a sum.  For
+    // DELTA kind this is expressed as the change in sum since the
+    // last collection.  For CUMULATIVE kind this is expressed as the
+    // last collected value of the sum.  For INSTANTANEOUS kind this
+    // is expressed as a scalar value that would be added into a Sum.
+    ADDING = 0x8;
+
+    // GROUPING structure means the value has been computed by
+    // combining individual values through a meaningful aggregation.
+    // GROUPING structure implies the sum of measurements is not
+    // necessarily meaningful.  These may be expressed as Histogram or
+    // Summary data points for DELTA kind and CUMULATIVE temporality
+    // kinds.  These may also be expressed as INSTANTANEOUS kind when
+    // reporting a scalar value type (e.g., for LastValue
+    // aggregation).  TODO(#159) or raw value (i.e., no aggregation).
+    GROUPING = 0x10;
+
+    // MONOTONIC may be set in conjunction with ADDING kinds.  When
+    // set, MONOTONIC indicates that the calculated sum is
+    // non-decreasing, therefore can be monitored as a non-negative
+    // rate of change.
+    MONOTONIC = 0x20;
```

Comment:
I read this, and I think we can better group all these 3 bits into an enum with 3 valid values:

```proto
enum Structure {
ADDING
MONOTONIC_ADDING
GROUPING
}
```

The main advantage is that we represent only valid combinations (no GROUPING_MONOTONIC, or at least not yet) and we use 1 less bit.

This also forces users when parsing the kind to switch on the 3 possible values for the "structure" property.

Reply:
Also no possible way to have ADDING_GROUPING (both bits set).

In terms of representing, even if we go with a bit for every possible combination (the current ADDING and GROUPING), we can replace the MONOTONIC bit to ADDING_MONOTONIC and achieve the same results as the enum, the main point is to unify the structure + monotonic concept into 3 possible values.

Reply:
This sounds good to me. I will apply this change.

Reply:
Done.